connector/datadogconnector/connector_native_test.go

```diff
@@ -268,3 +269,84 @@ func testMeasuredAndClientKindNative(t *testing.T, enableOperationAndResourceNam
 		t.Errorf("Diff between APM stats -want +got:\n%v", diff)
 	}
 }
+
+func TestObfuscate(t *testing.T) {
+	cfg := NewFactory().CreateDefaultConfig().(*Config)
+	cfg.Traces.BucketInterval = time.Second
+	connector, metricsSink := creteConnectorNativeWithCfg(t, cfg)
+
+	oconf := obfuscate.Config{Redis: obfuscate.RedisConfig{Enabled: true}}
+	connector.obfuscator = obfuscate.NewObfuscator(oconf)
+	connector.tcfg.Features["enable_receive_resource_spans_v2"] = struct{}{}
+	connector.tcfg.Features["enable_operation_and_resource_name_logic_v2"] = struct{}{}
```

Comment:
Prefer to set feature gates or connector configs than setting trace configs directly